### PR TITLE
[16.0][MIG] stock_picking_show_return: Migration to 16.0

### PR DESCRIPTION
**Pull request by slitheryduke11**

### CODE REVIEW

**Comment by slitheryduke11** [comment]
Hi @luisg123v, @pedrobaeza, @sergio-teruel, @bizzappdev . The migration of this module to 16.0 is ready.

I would really appreciate it if you could help me by reviewing the PR.

Thanks in advance!

**Comment by pedrobaeza** [comment]
/ocabot migration stock_picking_show_return

Please review other PRs, and in return, I will be glad to review yours.

**Review by bizzappdev — APPROVED** [review]
LGTM

**Review by desdelinux — APPROVED** [review]
LGTM

**Comment by desdelinux** [comment]
Functional test. Work as expected

@dreispt @pedrobaeza This is ready.

File: stock_picking_show_return/tests/test_stock_picking_show_return.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        self.product = self.env["product.product"].create({"name": "Test product"})
+        picking_type = self.env.ref("stock.picking_type_internal")
+        self.picking = self.env["stock.picking"].create(
+            {
+                "picking_type_id": picking_type.id,
+                "location_id": picking_type.default_location_src_id.id,
+                "location_dest_id": picking_type.default_location_dest_id.id,
+                "move_ids": [
+                    (
+                        0,
```
Comment (luisg123v) [inline stock_picking_show_return/tests/test_stock_picking_show_return.py]: Use Command syntax, please

File: stock_picking_show_return/models/stock_picking.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            picking.returned_ids = picking.mapped(
+                "move_ids.returned_move_ids.picking_id"
+            )
+
+    @api.depends("move_ids.origin_returned_move_id")
+    def _compute_source_picking_id(self):
+        """Get source piking from this picking. Only one origin is possible."""
+        for picking in self:
+            picking.source_picking_id = first(
+                picking.mapped("move_ids.origin_returned_move_id.picking_id")
```
Comment (luisg123v) [inline stock_picking_show_return/models/stock_picking.py]: Mapped not required anymore when the field is a recordset, see https://github.com/odoo/odoo/pull/32897

```suggestion
                picking.move_ids.origin_returned_move_id.picking_id
```

Please apply this in all similar cases

File: stock_picking_show_return/i18n/stock_picking_show_return.pot [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Translation of Odoo Server.
+# This file contains the translation of the following modules:
+# 	* stock_picking_show_return
+#
+msgid ""
+msgstr ""
+"Project-Id-Version: Odoo Server 15.0\n"
```
Comment (luisg123v) [inline stock_picking_show_return/i18n/stock_picking_show_return.pot]: 16.0.

Please re-generate this file (only the template)

File: stock_picking_show_return/tests/test_stock_picking_show_return.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
+# Copyright 2014-2017 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.tests.common import Form, TransactionCase, tagged
+
+
+@tagged("post_install", "-at_install")
+class TestStockPickingShowReturn(TransactionCase):
+    def setUp(self):
```
Comment (luisg123v) [inline stock_picking_show_return/tests/test_stock_picking_show_return.py]:
```suggestion
    @classmethod
    def setUpClass(cls):
```

**Review by luisg123v — CHANGES_REQUESTED** [review]

**Comment by slitheryduke11** [comment]
Hello @luisg123v.

All changes have been applied.

**Review by luisg123v — APPROVED** [review]
LGTM :+1: 

@moylop260 could you review/merge, please?

**Review by rousseldenis — APPROVED** [review]
Code review

**Comment by luisg123v** [comment]
Thanks @rousseldenis.

FYI when migrations are merged, that's usually done using nobump instead of patch, to make the version to stay in 16.0.1.0.0

**Comment by rousseldenis** [comment]
> Thanks @rousseldenis.
> 
> FYI when migrations are merged, that's usually done using nobump instead of patch, to make the version to stay in 16.0.1.0.0

Yes from >= 15, pre-releases are not done anymore on pypi. I forgot that information. That's why I'd always prefer use patch before. So, usage of nobump is encouraged now

**Pull request merged**